### PR TITLE
[PREGEL] use general messages for all communication between conductor and workers

### PR DESCRIPTION
**Pull request by jvolmer**
Use the general message for all communication between conductor and workers.

### CODE REVIEW

File: arangod/Pregel/Conductor/Conductor.cpp [hunk 1]
```diff
@@ -231,17 +255,17 @@ auto Conductor::_preGlobalSuperStep() -> bool {
...
-void Conductor::workerStatusUpdate(VPackSlice const& data) {
+void Conductor::workerStatusUpdate(StatusUpdated const& data) {
   MUTEX_LOCKER(guard, _callbackMutex);
   // TODO: for these updates we do not care about uniqueness of responses
   // _ensureUniqueResponse(data);
 
-  auto event = deserialize<StatusUpdated>(data);
-
+  VPackBuilder event;
+  serialize(event, data);
```
Comment (romanatarango) [inline arangod/Pregel/Conductor/Conductor.cpp:264]: This is only needed in debug mode.

File: arangod/Pregel/OutgoingCache.cpp [hunk 2]
```diff
@@ -100,50 +102,51 @@ void ArrayOutCache<M>::flushMessages() {
...
+        {
+          VPackArrayBuilder ab2(&messagesVPack);
+          for (M const& message : messages) {
+            this->_format->addValue(messagesVPack, message);
+            if (this->_sendToNextGSS) {
+              this->_sendCountNextGSS++;
+            } else {
+              this->_sendCount++;
+            }
+          }
```
Comment (romanatarango) [inline arangod/Pregel/OutgoingCache.cpp:129]: Can we use `serialize` for this?

**Review by romanatarango — APPROVED** [review]
LGTM, see comments about small things.

File: arangod/Pregel/OutgoingCache.cpp (hunk 2 above)
Comment (jvolmer) [reply arangod/Pregel/OutgoingCache.cpp:129]: Yes, I would like to, but it is a bit more complicated, therefore I would like to do it in a separate PR: 
Messages are of generic type M. Each algorithm defines the type M and uses a Format class of type M to do serialization and deserialization. Therefore in the PregelMessage struct we have to make sure that we always use the correct message type.
Additionally we have different types of things we send around: either an array of messages to one vertex (using ArrayOutCache and ArrayInCache) or one message to one vertex (using CombiningOutCache and CombiningInCache). This difference has to be taken into account inside the PregelMessage if we don't use a basic VPackBuilder.
We should definitely do that, but I think it is a bit more work and should be done in a separate PR.

File: arangod/Pregel/Conductor/Conductor.cpp (hunk 1 above)
Comment (jvolmer) [reply arangod/Pregel/Conductor/Conductor.cpp:264]: What is your suggestion here?

**Pull request merged**